R/uvhydrograph-render.R

```diff
@@ -45,11 +47,25 @@ createPrimaryPlot <- function(data, month){
     
     correctionLabels <- parseLabelSpacing(primaryData[i], primaryInfo)
     primaryApprovals <-  parseApprovalInfo(primaryData[i], primaryInfo, x, y)
-    primaryStyles <- getUvStyle(primaryData[i], primaryInfo, x, y, primaryApprovals, correctionLabels, "primary")
-    primaryPlotTypes <- getPlotType(primaryData[i], "primary")
-   
-    if (names(primaryData[i]) %in% c("series_corr", "meas_Q")) {
-      for (j in 1:length(primaryStyles)) {
+    
+    if (length(primaryApprovals) > 0) {
+      primaryStyles <- vector("list", length(primaryApprovals))
+      primaryPlotTypes <- c()
+      notPlot <- c()
+      for (k in seq(length(primaryApprovals))) {
+        primaryStyles[[k]] <- getUvStyle(primaryData[i], primaryInfo, x, y, primaryApprovals[[k]], correctionLabels, "primary")
+        primaryPlotTypes[k] <- getPlotType(primaryData[i], "primary")
+        notPlot[k] <- all(is.na(primaryStyles[[k]]$x))
+      }
+      primaryStyles[which(notPlot)] <- NULL
+      primaryPlotTypes <- primaryPlotTypes[which(!notPlot)]
+    } else {
+      primaryStyles <- getUvStyle(primaryData[i], primaryInfo, x, y, primaryApprovals, correctionLabels, "primary")
+      primaryPlotTypes <- getPlotType(primaryData[i], "primary")
+    }
+    
+    if (names(primaryData[i]) %in% c("series_corr", "meas_Q", "max_DV", "min_DV", "median_DV", "mean_DV", "UV_series")) {
+      for (j in seq(length(primaryStyles))) {
```

Comment:
Ah good catch, I do this in a couple of places so I'll make sure to change it. Thanks!